### PR TITLE
W-10875597 - W-10854067 / Recurring donation design - donation type filter for admin

### PR DESCRIPTION
**Pull request by maurimart**
# Critical Changes

Create Recurring donation table component
Configuration of metadata of Recurring donation table component to expose donation type filter for admins

# Changes

Add custom labels for Recurring donation table columns

### CODE REVIEW

**Comment by maurimart** [comment]
[W-10875597](https://gus.lightning.force.com/a07EE00000tMRzhYAG)
[W-10854067](https://gus.lightning.force.com/a07EE00000t34xBYAQ)

**Review by JoseRossiSF — APPROVED** [review]
LGTM

**Review by gerardorodriguezsf — APPROVED** [review]
LGTM

File: force-app/main/default/lwc/recurringDonationTable/recurringDonationTable.js-meta.xml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<LightningComponentBundle xmlns="http://soap.sforce.com/2006/04/metadata">
+    <apiVersion>53.0</apiVersion>
+    <isExposed>true</isExposed>
+    <masterLabel>Recurring Donations Table</masterLabel>
```
Comment (gerardorodriguezsf) [inline force-app/main/default/lwc/recurringDonationTable/recurringDonationTable.js-meta.xml:5]: @maurimart According to the ticket AC the component name should be "Recurring Donations".

File: force-app/main/default/lwc/recurringDonationTable/recurringDonationTable.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        { label: 'View', name: 'view' },
+        { label: 'Delete', name: 'delete' }
+    ];
+    columns = [
+        { label: commonAmount, fieldName: "amount", type: "currency", hideDefaultActions: true },
+        { label: RDCL_Frequency, fieldName: "frequency", type: "text", hideDefaultActions: true },
+        { label: commonPaymentMethod, fieldName: "paymentMethod", type: "text", hideDefaultActions: true },
+        { label: lblStatus, fieldName: "status", type: "text", hideDefaultActions: true },
+        { label: dateStarted, fieldName: "dateStarted", type: "text", hideDefaultActions: true },
+        { label: lastDonationDate, fieldName: "lastDonationDate", type: "text", hideDefaultActions: true },
```
Comment (gerardorodriguezsf) [inline force-app/main/default/lwc/recurringDonationTable/recurringDonationTable.js:85]: @maurimart Missing "End Date" column.

File: force-app/main/default/lwc/recurringDonationTable/recurringDonationTable.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+import dateStarted from '@salesforce/label/c.dateStarted'
+import lastDonationDate from '@salesforce/label/c.lastDonationDate';
+import nextInstallment from '@salesforce/label/c.nextInstallment';
+import lastUpdate from '@salesforce/label/c.lastUpdate';
+import commonActions from '@salesforce/label/c.commonActions';
+
+export default class RecurringDonationTable extends LightningElement {
+    @api
+    donationTypeFilter = 'all';
+    data = [{
```
Comment (gerardorodriguezsf) [inline force-app/main/default/lwc/recurringDonationTable/recurringDonationTable.js:15]: @maurimart Add some string values to the "Next Installment" mock data, fix date format.

**Review by gerardorodriguezsf — CHANGES_REQUESTED** [review]

File: force-app/main/default/lwc/recurringDonationTable/recurringDonationTable.js-meta.xml (hunk 1 above)
Comment (maurimart) [reply force-app/main/default/lwc/recurringDonationTable/recurringDonationTable.js-meta.xml]: Done!

File: force-app/main/default/lwc/recurringDonationTable/recurringDonationTable.js (hunk 2 above)
Comment (maurimart) [reply force-app/main/default/lwc/recurringDonationTable/recurringDonationTable.js:92]: Done!

**Review by gerardorodriguezsf — APPROVED** [review]

File: force-app/main/default/labels/CustomLabels.labels-meta.xml [hunk 4]
```diff
@@ -5864,6 +5864,62 @@ To check your GAU Allocation settings, go to Donations | GAU Allocations.</value
...
         <value>Amount</value>
     </labels>
+    <labels>
+        <fullName>commonPaymentMethod</fullName>
+        <categories>Gift Entry</categories>
+        <language>en_US</language>
+        <protected>true</protected>
+        <shortDescription>Payment Method</shortDescription>
+        <value>Payment Method</value>
+    </labels>
```
Comment (bdvorachek) [inline force-app/main/default/labels/CustomLabels.labels-meta.xml:5874]: I don't think we need to add a label for "Payment Method". Can we use the field label from the Recurring Donation object's Payment Method field instead?

File: force-app/main/default/labels/CustomLabels.labels-meta.xml [hunk 5]
```diff
@@ -5864,6 +5864,62 @@ To check your GAU Allocation settings, go to Donations | GAU Allocations.</value
...
+        <shortDescription>Date Started</shortDescription>
+        <value>Date Started</value>
+    </labels>
+    <labels>
+        <fullName>lastDonationDate</fullName>
+        <categories>Gift Entry</categories>
+        <language>en_US</language>
+        <protected>true</protected>
+        <shortDescription>Last Donation Date</shortDescription>
+        <value>Last Donation Date</value>
```
Comment (bdvorachek) [inline force-app/main/default/labels/CustomLabels.labels-meta.xml:5889]: Can we use the npe03__Last_Payment_Date__c field on the Recurring Donations object for this label instead? The label for that field matches up with what's in the custom label here, "Last Donation Date".

**Review by bdvorachek — CHANGES_REQUESTED** [review]
There's two custom labels here that are duplicates of some field labels on the RD object. Whenever it is possible, we should re-use those field labels.

File: force-app/main/default/labels/CustomLabels.labels-meta.xml (hunk 4 above)
Comment (maurimart) [reply force-app/main/default/labels/CustomLabels.labels-meta.xml:5874]: good suggestion

File: force-app/main/default/labels/CustomLabels.labels-meta.xml (hunk 5 above)
Comment (maurimart) [reply force-app/main/default/labels/CustomLabels.labels-meta.xml:5889]: done

**Review by bdvorachek — APPROVED** [review]

**Pull request merged**